Mathlib/FieldTheory/PrimitiveElement.lean

```diff
@@ -348,10 +347,82 @@ end FiniteIntermediateField
 
 end Field
 
+variable (F E : Type*) [Field F] [Field E] [Algebra F E] [FiniteDimensional F E] [IsSeparable F E]
+
 @[simp]
-theorem AlgHom.card (F E K : Type*) [Field F] [Field E] [Field K] [IsAlgClosed K] [Algebra F E]
-    [FiniteDimensional F E] [IsSeparable F E] [Algebra F K] :
+theorem AlgHom.card (K : Type*) [Field K] [IsAlgClosed K] [Algebra F K] :
     Fintype.card (E →ₐ[F] K) = finrank F E := by
   convert (AlgHom.card_of_powerBasis (L := K) (Field.powerBasisOfFiniteOfSeparable F E)
     (IsSeparable.separable _ _) (IsAlgClosed.splits_codomain _)).trans (PowerBasis.finrank _).symm
 #align alg_hom.card AlgHom.card
+
+@[simp]
+theorem IsNormalClosure.algHom.card (L : Type*) [Field L] [Algebra F L] [IsNormalClosure F E L]  :
+    Fintype.card (E →ₐ[F] L) = FiniteDimensional.finrank F E := by
+  rw [← Fintype.ofEquiv_card <|
+    IsNormalClosure.algHom_equiv_algHom_isAlgClosed F E L (AlgebraicClosure F)]
+  convert AlgHom.card F E (AlgebraicClosure F)
+
+section iff
+
+namespace Field
+
+open FiniteDimensional IntermediateField Polynomial Algebra Set
+
+variable (F : Type*) {E : Type*} [Field F] [Field E] [Algebra F E] [FiniteDimensional F E]
+
+theorem primitive_element_of_minpoly_natDegree_eq (α : E) :
+    F⟮α⟯ = ⊤ ↔ (minpoly F α).natDegree = finrank F E := by
+  rw [← adjoin.finrank (IsIntegral.of_finite F α), ← finrank_top F E]
+  refine ⟨fun h => ?_, fun h => ?_⟩
+  · exact congr_arg (fun K : IntermediateField F E => finrank F K) h
+  · exact eq_of_le_of_finrank_eq le_top h
+
+theorem primitive_element_of_minpoly_degree_eq (α : E) :
+    F⟮α⟯ = ⊤ ↔ (minpoly F α).degree = finrank F E := by
+  rw [degree_eq_iff_natDegree_eq, primitive_element_of_minpoly_natDegree_eq]
+  exact minpoly.ne_zero_of_finite F α
+
+variable [IsSeparable F E] (A : Type*) [Field A] [Algebra F A] [hA : IsNormalClosure F E A]
+
+theorem primitive_element_iff_algHom_eq_of_eval' (α : E) :
+    F⟮α⟯ = ⊤ ↔ ∀ φ ψ : E →ₐ[F] A, φ α = ψ α → φ = ψ := by
```

Comment:
Is it better to write `F⟮α⟯ = ⊤ ↔ Function.Injective fun (φ : E →ₐ[F] A) => φ α`? (untested)